virtualizarr/tests/test_readers/test_safetensors.py

```diff
@@ -0,0 +1,348 @@
+            "weight": np.random.randn(10, 20).astype(np.float32),
+            "bias": np.random.randn(20).astype(np.float32),
+        }
+
+        # Convert complex nested metadata to strings as required by safetensors
+        model_info = {
+            "name": "test_model",
+            "version": "1.0",
+            "parameters": 220,
+            "layers": [
+                {"name": "layer1", "type": "linear"},
+                {"name": "layer2", "type": "activation"},
+            ],
+        }
+        training = {"epochs": 10, "optimizer": "adam", "learning_rate": 0.001}
+
+        metadata = {
+            "model_info": json.dumps(model_info),
+            "training": json.dumps(training),
+        }
+
+        save_file(tensors, filepath, metadata=metadata)
+
+    yield filepath
+
+    os.unlink(filepath)
+
+
+@pytest.mark.skipif(not HAS_SAFETENSORS, reason="safetensors not installed")
+def test_reader_registration():
+    """Test that the SafeTensors reader is properly registered in VirtualiZarr."""
+    from virtualizarr.backend import VIRTUAL_BACKENDS
+
+    assert "safetensors" in VIRTUAL_BACKENDS
+    assert VIRTUAL_BACKENDS["safetensors"] == SafeTensorsVirtualBackend
+
+
+@pytest.mark.skipif(not HAS_SAFETENSORS, reason="safetensors not installed")
+def test_file_detection(sample_safetensors_file):
+    """Test that VirtualiZarr correctly identifies the file as a SafeTensors file."""
+    from virtualizarr.backend import automatically_determine_filetype
+
+    filetype = automatically_determine_filetype(filepath=sample_safetensors_file)
+    assert filetype == FileType.safetensors
+
+
+@pytest.mark.skipif(not HAS_SAFETENSORS, reason="safetensors not installed")
+def test_open_virtual_dataset(sample_safetensors_file):
+    """Test opening a SafeTensors file as a virtual dataset."""
+    vds = open_virtual_dataset(sample_safetensors_file)
+
+    assert set(vds.variables) == {"tensor1", "tensor2", "tensor3"}
+
+    assert vds["tensor1"].shape == (10, 10)
+    assert vds["tensor2"].shape == (5, 5)
+    assert vds["tensor3"].shape == (10, 10)
+
+
+@pytest.mark.skipif(not HAS_SAFETENSORS, reason="safetensors not installed")
+def test_tensor_values(sample_safetensors_file):
+    """Test that the tensor values are correctly read."""
+
+    loadable_variables = ["tensor1", "tensor2", "tensor3"]
+    vds = open_virtual_dataset(
+        sample_safetensors_file, loadable_variables=loadable_variables
+    )
+
+    tensor1 = vds["tensor1"].values
+    tensor2 = vds["tensor2"].values
+    tensor3 = vds["tensor3"].values
+
+    expected_tensor1 = np.ones((10, 10))
+    expected_tensor2 = np.zeros((5, 5))
+    expected_tensor3 = np.arange(100).reshape(10, 10)
+
+    np.testing.assert_array_equal(tensor1, expected_tensor1)
+    np.testing.assert_array_equal(tensor2, expected_tensor2)
+    np.testing.assert_array_equal(tensor3, expected_tensor3)
+
+
+@pytest.mark.skipif(not HAS_SAFETENSORS, reason="safetensors not installed")
```

Comment:
We prefer to generalize this pattern as `requires_safetensors`, see https://github.com/zarr-developers/VirtualiZarr/blob/d6b404d5ffd840f31d5517a6184863ff57c433a9/virtualizarr/tests/__init__.py#L33